### PR TITLE
🏗️: use src-layout

### PR DESCRIPTION
**Pull request by nstarman**
I haven't used setuptools in a while, let's see what the CI thinks.

### CODE REVIEW

**Comment by nstarman** [comment]
Thanks! This isnt a concern when using hatch & hatch-vcs. Maybe I'll start by pushing a PR to upgrade the build system.

**Comment by mhvk** [comment]
If it is not essential, I'd prefer to just stick with things that astropy uses as much as possible. It is something we all presumably use regularly, so provides some common denominator.

p.s. The initial setup from @adrn turned out to be something I had no idea how it worked, and I really would like not to spent even more of my time (and astropy money) on things like that! It is bad enough that numpy uses a completely different system (even if understandable, given that they need to deal with different compilers, etc.); please no more!

**Comment by nstarman** [comment]
> If it is not essential, I'd prefer to just stick with things that astropy uses as much as possible.

Fair enough. But if we want to speed up this library considerably using e.g. mypyc I was planning to copy-paste from `black` -- https://github.com/psf/black/blob/main/pyproject.toml — who've thought about this extensively. When I looked a few years ago Hatch didn't used to support the mypyc use case. Now it's the preferred solution for pure Python installs and associated transpilation.

**Comment by mhvk** [comment]
I now looked in a bit more detail: the annoyance is that `hatch` is not packaged by debian yet [1] so it means that to even start developing I need to enter a virtual environment. And it looks like it is meant to be a complete replacement for `tox`... But obviously your time is worth a lot too, so if that will make other things much easier, I can adapt.

[1] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1072818

**Comment by nstarman** [comment]
Thankfully we only need hatchling! https://hatch.pypa.io/1.9/install/#build-system-availability

**Comment by mhvk** [comment]
`--pyargs` takes a package name as its argument -- I think that should still be `quantity`

File: tox.ini [hunk 1]
```diff
@@ -58,8 +58,8 @@ commands_pre =
     pip freeze
 
 commands =
-    !cov: pytest --pyargs quantity {posargs} {toxinidir}/tests {toxinidir}/docs
-    cov: pytest --pyargs quantity --cov quantity --cov-report xml:coverage.xml --cov-report term-missing {posargs} {toxinidir}/tests {toxinidir}/docs
+    !cov: pytest --pyargs src {posargs} {toxinidir}/tests {toxinidir}/docs
```
Comment (mhvk) [inline tox.ini:61]: Locally, if I change this and the one below to `quantity`, things work.

Comment (nstarman) [reply tox.ini]: Thanks! Made the change.

File: tox.ini [hunk 2]
```diff
@@ -59,7 +59,7 @@ commands_pre =
 
 commands =
     !cov: pytest --pyargs quantity {posargs} {toxinidir}/tests {toxinidir}/docs
-    cov: pytest --pyargs quantity --cov quantity --cov-report xml:coverage.xml --cov-report term-missing {posargs} {toxinidir}/tests {toxinidir}/docs
+    cov: pytest --pyargs quantity --cov src --cov-report xml:coverage.xml --cov-report term-missing {posargs} {toxinidir}/tests {toxinidir}/docs
```
Comment (mhvk) [inline tox.ini:62]: This one also should be just `quantity` (tested locally; we don't have a coverage test run on CI yet). I.e., `tox.ini` can remain unchanged.

**Review by mhvk — APPROVED** [review]
This looks good to me.

**Pull request merged**